include/envoy/api/io_error.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/platform.h"
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Api {
+
+class IoError;
+
+// IoErrorCode::Again is used frequently. Define it to be a distinguishable address to avoid
+// frequent memory allocation of IoError instance.
+// If this is used, IoCallResult has to be instantiated with a deleter that does not
+// deallocate memory for this error.
+#define ENVOY_ERROR_AGAIN reinterpret_cast<Api::IoError*>(0x01)
```

Comment:
This might be changed in #6037 to address an memory alignment error.

Reply:
Oh good :)  Can I suggest addressing this with a singleton? I'm generally against singletons but this seems like a good place to use one.

I'll comment also in #6037 .

Reply:
We'll let that be handled in #6037 .